### PR TITLE
Add support for Ubuntu 22.04, Fedora 36, EL 9, openSUSE 15.4

### PR DESCRIPTION
**Pull request by pstorz**
This PR enables building for Ubuntu 22.04, Fedora 36, EL 9, openSUSE 15.4.
Removing support for obsolete openSUSE 15.2 and Fedora 34.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

##### Tests

- [x] Tests are runs on new platform

### CODE REVIEW

**Review by bruno-at-bareos — CHANGES_REQUESTED** [review]
Mostly okay, see proposal for the suse gcc block in spec.

**Comment by bruno-at-bareos** [comment]
During the review, the check-sources return modification on 
```

File: core/platforms/packaging/bareos.spec [hunk 1]
```diff
@@ -122,13 +123,18 @@ BuildRequires: devtoolset-8-gcc
...
+BuildRequires: gcc11-c++
+%else
+  %if 0%{?sle_version} == 150300 || 0%{?suse_version} > 1500
 BuildRequires: gcc10
 BuildRequires: gcc10-c++
-%else
-  %if 0%{?suse_version}
+  %else
+    %if 0%{?suse_version}
 BuildRequires: gcc9
```
[the remark this answers is not in the document]
Comment (pstorz) [reply core/platforms/packaging/bareos.spec:135]: As we talked, we do not want to change the compiler of a bareos release for a specific OS release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ and since Bareos version 20 this project adheres to [Semantic Versioning](https:
 - systemtest py*plug-fd-contrib-bareos_tasks_mysql [PR #768]
 - webui: introduce rerun of multiple jobs at once [PR #1109]
 - dird: console: add the ability to rerun multiple commas separated jobids [PR #1170]
+- build: Add support for Ubuntu 22.04, Fedora 36, EL 9, openSUSE 15.4
```
Comment (bruno-at-bareos) [inline CHANGELOG.md:32]:
```suggestion
- build: Add support for Ubuntu 22.04, Fedora 36, EL 9, openSUSE 15.4 [PR #1179]
```

**Review by bruno-at-bareos — APPROVED** [review]
After last green, the PR is ready to be merged. Thanks for the work.

**Pull request merged**